crates/api/src/instance.rs

```diff
@@ -58,6 +58,13 @@ pub struct Instance {
 }
 
 impl Instance {
+    pub fn set_signal_handler<H>(&mut self, handler: H)
```

Comment:
It'd be good to have a documentation comment here, explaining what this is for and what the requirements of the handler are. In particular, we should describe the use of the return value.